### ISSUE TITLE
Feature Request: Data source to discover all VPC IDs

### ISSUE THREAD

**Issue by erikpaasonen** [terraform-providers/terraform-provider-aws#3984]
Would like to see support for an `aws_vpc_ids` data source which outputs all VPC IDs in the same manner that `aws_subnet_ids` outputs all subnet IDs for a given VPC. The primary use case would interpolate the output into `count` of an `aws_flow_log` resource.

Should work without any arguments; the provider will handle account and region context.

Equivalent of running `aws ec2 describe-vpcs` and selecting the `VpcId` field.

### Suggested Syntax
``` HCL
data "aws_vpc_ids" "foo" {}

output "foo" {
  value = "${data.aws_vpc_ids.foo.ids}"
}

resource "aws_flow_log" "test_flow_log" {
  count = "${length(data.aws_vpc_ids.foo.ids)}"
  ...
  vpc_id = "${element(data.aws_vpc_ids.foo.ids, count.index)}"
  ...
}
```

### Expected Output
With 2 VPCs present:
``` HCL
foo = [
  vpc-a1b2c3d,
  vpc-e4d5f6g
]
```

With zero VPCs:
``` HCL
foo = []
```

### References
https://docs.aws.amazon.com/cli/latest/reference/ec2/describe-vpcs.html

**Comment by bflad** [comment]
A new `aws_vpcs` data source will be available in version 1.23.0 of the AWS provider, likely releasing later today.

**Comment by bflad** [comment]
This has been released in [version 1.23.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1230-june-14-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
